modules/ROOT/pages/hash-client-credentials.adoc

```diff
@@ -0,0 +1,40 @@
+= Configure Credentials Hashing for Mule 4
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+:keywords: hashing, client credentials, auto-generated, raml, wsdl, apikit, http
+
+You enable hashing of your client credentials to add another layer of security for your APIs by hiding the client secrets from the Mule runtime engine (Mule) server. When you enable client hashing, API gateway stores the client credentials in memory, with only the client secret part hashed. 
+
+Additionally, if you enabled both encryption and hashing in API gateway, the contract information in memory is hashed, but not encrypted. When client hashing (disabled by default) is enabled, it uses the SHA-256 hashing algorithm.
+
+If you modify the configuration to enable or disable API gateway for hashing and restart your Mule server for the updated configuration to apply, your incoming requests are not affected and they continue to be processed normally. 
```

Comment:
```suggestion
If you modify the configuration to enable or disable hashing in API gateway and then restart your Mule server to apply the updated configuration, your incoming requests are not affected and they continue to be processed normally. 
```

Reply:
Question: if restarting applies the update, why does that not affect processing behavior from that point?

Reply:
Done! Thanks!